cdk/kittyhawk/src/ingress.ts

```diff
@@ -15,7 +15,24 @@ export interface IngressProps {
      *
      * @default undefined
      */
-  readonly ingress?: { host: string; paths: string[] }[];
+  readonly ingress?: HostRules[];
```

Comment:
Is there a reason that ingress is an optional prop? I feel like if you want to make an ingress then you would pretty much have to define it

Reply:
it's set this way because I defined ApplicationProps as
```
export interface ApplicationProps extends IngressProps, DeploymentProps,
  ServiceProps { }
```
but not all applications will have an ingress. 

Can you think of a better way to do this? I tried doing Partial< IngressProps> but it still causes problems since it fails the type checks if I try to pass ApplicationProps to IngressProps.

Reply:
Hmm, it looks like `port` is the only field that shared between any of the base props, right? I personally would probably define ApplicationProps something like this:

```javascript
interface ApplicationProps = {
  ingress: IngressProps
  service: ServiceProps
  deployment: DeploymentProps
  servicePort: number
}
```
and then just pass through the props to their respective components while also injecting the servicePort into both the ingress and service objects. 

I personally like having props set up this way (I used this format for kraken) because it makes it super clear what keys fall under which objects. So even if I don't know too much about k8s, I know that a field named `containerPort` is under the service props, so it must relate to that